### ISSUE TITLE
aws_eks_fargate_profile inconsistent result

### ISSUE THREAD

**Issue by ivankatliarchuk** [hashicorp/terraform-provider-aws#13372]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

0.12.25

### Affected Resource(s)

aws_eks_fargate_profile

* aws_XXXXX

### Terraform Configuration Files

```hcl
resource aws_eks_fargate_profile first {
  cluster_name           = "cluster"
  fargate_profile_name    = "profile-furst"
  pod_execution_role_arn = "arn-first"
  subnet_ids             = ["subnets"]

  selector {
    namespace = "default"
  }
}

resource aws_eks_fargate_profile second {
  cluster_name           = "cluster"
  fargate_profile_name    = "profile-second"
  pod_execution_role_arn = "arn-second"
  subnet_ids             = ["subnets"]

  selector {
    namespace = "default"
  }
}
```

### Panic Output

```
Error: error creating EKS Fargate Profile (cluster:profile-second): ResourceInUseException: Cannot create Fargate Profile profile-second because cluster cluster currently has Fargate profile profile-first in status CREATING
{
  RespMetadata: {
    StatusCode: 409,
    RequestID: "0fd17f34-b86e-449a-a685-a77ed0a3f3c6"
  },
  ClusterName: "cluster",
  Message_: "Cannot create Fargate Profile `profile-second` because cluster `cluster` currently has Fargate profile -profile-first- in status CREATING"
}
```

### Expected Behavior

The profile creation should wait, since the error message clearly shows there is profile currently creationg

### Actual Behavior

Error. Cannot create multiple profiles

### Steps to Reproduce

run the config

1. `terraform apply`

### References

* #0000

**Comment by ivankatliarchuk** [comment]
IAM roles as in docs : https://www.terraform.io/docs/providers/aws/r/eks_fargate_profile.html

**Comment by bflad** [comment]
The fix for this has been merged for both creation and deletion cases. It will be released with version 3.16.0 of the Terraform AWS Provider, likely later today or tomorrow. Thank you to @jhujasonw for the implementation. 👍
